src/logic/currentSession/store/reducer/currentSession.ts

```diff
@@ -22,13 +25,14 @@ export default handleActions<AppReduxState['currentSession'], CurrentSessionPayl
     [LOAD_CURRENT_SESSION]: (state = initialState, action: Action<CurrentSessionState>) => ({
       ...state,
       ...action.payload,
+      restored: true,
     }),
     [UPDATE_VIEWED_SAFES]: (state, action: Action<string>) => {
       const safeAddress = action.payload
-      const viewedSafes = state.viewedSafes
+      const viewedSafes = state.viewedSafes.filter((item) => item !== safeAddress)
       const newState = {
         ...state,
-        viewedSafes: viewedSafes.includes(safeAddress) ? viewedSafes : [...viewedSafes, safeAddress],
+        viewedSafes: [safeAddress].concat(viewedSafes).slice(0, maxViewedSafes),
```

Comment:
I've changed the logic here a bit, to keep the list in a chronological order. It wasn't used anywhere as far as I can tell.